src/WebJobs.Script/Config/HostJsonFileConfigurationOptions.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    public class HostJsonFileConfigurationOptions
+    {
+        private const string ConfigProfileKey = "configurationProfile";
+        private const string ConfigProfileEnvKey = $"{ConfigurationSectionNames.JobHost}__{ConfigProfileKey}";
+
+        private string _configProfile;
```

Comment:
Using a constructor, this could be made readonly, right? That would avoid potential mutation in future changes, which could lead to unintended behavior (e.g., different results across multiple calls of GetConfigProfile)

Reply:
Switched to a ctor.